include/CLI/App.hpp

```diff
@@ -147,6 +147,12 @@ class App {
     /// @name Help
     ///@{
 
+    /// Usage to put after program/subcommand description in the help output INHERITABLE
+    std::string usage_{};
```

Comment:
I find it confusing that there are two different things here, both set from the same function, that will both potentially show up. I think it would be better if there was just a `usage_callback_`, and `usage(std::string usage_string)` would just call usage again withset `[](){return usage_string;}`. Simpler internals, and I think a lot more predictable usage.

Reply:
I just followed `footer` implementation as example when implementing similar behavior for `usage`.
So, if you think it's better to use only `callback` for `usage`, then I'll make the appropriate changes.

Could you clarify, should I leave only `callback` as you described?
Should appropriate changes be made for `footer`?

Reply:
The footer has two use cases.  One is to supply additional information via the callback.  I use this to trigger help in cascading apps.  The callback is not inherited.  The other case is to supply branding or copyright information which does need to be inherited(or at least can be) so it is printed on all subcommands, and both are used when generating the help.  

I am not sure the same situation applies to the usage, also not clear that usage should be inherited like the footer.  So it might make sense to not exactly follow the pattern for the footer.

Reply:
See more discussion in, too: https://github.com/CLIUtils/CLI11/pull/309#issuecomment-522364430.

Reply:
@phlptp, thanks for the clarification, now I get it.
Yes, I'll fix that behavior.

Reply:
@henryiii, @phlptp,  little clarification to see if I understood correctly.

You suggest using only callback, something like this:

```c++
    /// Set usage.
    App *usage(std::string usage_string) {
        usage_callback_ = std::move([usage_string](){return usage_string;});
        return this;
    }
    /// Set usage.
    App *usage(std::function<std::string()> usage_function) {
        usage_callback_ = std::move(usage_function);
        return this;
    }
    /// Generate and return the usage.
    CLI11_NODISCARD std::string get_usage() const {
        return (usage_callback_) ? usage_callback_() : "";
    }
```

Without using separate string variable for plain text.
So, in this case, `usage` setters (with a string parameter and with a callback function parameter) will be mutually exclusive and non-inherited.

If so, I can make changes within a few hours.

Reply:
Yes, I think that's the idea.